### DISCUSSION TITLE
Debug log to a file

### DISCUSSION THREAD

**Discussion by BobanSpasic** [rsta2/circle#430] [Q&A]
Hi folks

is there any possibility to get a debug log as a file?
Namely, I am trying to get MiniDexed to run on Zero 2W in USB Gadget mode. I have an HDMI monitor (7" touch), but it does not show anything (it goes to standby) so I have no clue why it isn't booting. It would probably help if I could get some debug log in a file.

**Comment by rsta2** [comment]
There is no standard way for writing a log file, but with some small source code it is possible. See the archive attached. It contains two files *fatfslogfile.h* and *fatfslogfile.cpp*, which must be copied into the *src/* directory of MiniDexed. Then apply the patch *logfile.patch* from MiniDexed root with
`patch -p1 < logfile.patch` and rebuild MiniDexed. I did not test it with MiniDexed, so problems may happen.

It should open a file *minidexed.log* on the SD card and writes eventual log events into it. Please note that the file may remain open, because there is nothing (despite a hard system error), which can trigger the close. You may have to run a file system check (e.g. *fsck.vfat*) for the SD card after running it. That's not perfect, but for a quick test, it should be OK.

There is some chance, that your problem happens, because the via USB supplied power is not sufficient. In this case this tool will not help, because the system does not run at all.

**Comment by rsta2** [comment]
Oops, I forgot the archive:

[minidexed-logfile.zip](https://github.com/rsta2/circle/files/14324769/minidexed-logfile.zip)

**Comment by BobanSpasic** [comment]
Did some further tests here https://github.com/probonopd/MiniDexed/discussions/568#discussioncomment-8508592
Raspbian Lite (console only) 32-bit boots fine. LibreELEC stops with an error.
I'll see if someone could build the version with logging for me because I have no clue how to do it myself.
